pyi.py

```diff
@@ -425,6 +427,60 @@ def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
     def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef) -> None:
         self._visit_function(node)
 
+    def _Y019_error(
+        self, node: ast.FunctionDef | ast.AsyncFunctionDef, typevar_name: str
+    ) -> None:
+        error_message = Y019.format(typevar_name=typevar_name)
+
+        if sys.version_info >= (3, 9):
+            cleaned_method = deepcopy(node)
+            cleaned_method.decorator_list.clear()
+            new_syntax = re.sub(
+                fr"(\W){typevar_name}(\W)", r"\1Self\2", ast.unparse(cleaned_method)
+            ).replace("\n   ", "")
```

Comment:
This seems fragile, as it assumes you indent with 4 spaces. It's common, but not required. The following replaces all whitespace characters, including newlines and tabs, with a single space:

```suggestion
            )
            new_syntax = re.sub(r"\s+", " ", new_syntax)
```

Reply:
Also, classes can be nested inside `if sys.version_info` checks and so methods can be indented with 8 spaces, even in blacked code.